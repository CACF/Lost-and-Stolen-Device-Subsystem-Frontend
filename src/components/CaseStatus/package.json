{
  "name": "CaseStatus",
  "version": "0.0.0",
  "private": true,
  "main": "./CaseStatus.js"
}
