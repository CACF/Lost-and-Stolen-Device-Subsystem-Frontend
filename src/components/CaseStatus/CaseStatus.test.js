import React from 'react';
import {mount, shallow} from 'enzyme';
import CaseStatus from './CaseStatus';
import { BrowserRouter as Router } from 'react-router-dom';

const location = {
  pathname: '/case-status/',
  state: {
    details: {
      icon: 'fa fa-check',
      action: 'Register',
      id: 12,
      status: 'Pending'
    }
  }
};
describe('CaseStatus Components', ()=> {
  test("if renders correctly", () => {
    const wrapper = shallow(
      <Router>
        <CaseStatus location={location}/>
      </Router>);
    expect(wrapper).toMatchSnapshot();
  });

  test('link should be wrapped in specific class', () => {
    const wrapper = mount(
      <Router>
        <CaseStatus location={location}/>
      </Router>);
    expect(wrapper.find('.link-box')).toHaveLength(1);
  });

  test('should have a single link', () => {
    const wrapper = mount(
      <Router>
        <CaseStatus location={location}/>
      </Router>);
    expect(wrapper.find('a')).toHaveLength(1);
  });

  test('should render icon correctly', () => {
    const wrapper = mount(
      <Router>
        <CaseStatus location={location}/>
      </Router>);
    expect(wrapper.find('.icon-box').find('i').hasClass('fa fa-check')).toBe(true);
  });
});