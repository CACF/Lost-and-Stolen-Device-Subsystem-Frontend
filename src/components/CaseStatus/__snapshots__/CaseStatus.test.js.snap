// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseStatus Components if renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <CaseStatus
      location={
        Object {
          "pathname": "/case-status/",
          "state": Object {
            "details": Object {
              "action": "Register",
              "icon": "fa fa-check",
              "id": 12,
              "status": "Pending",
            },
          },
        }
      }
    />
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <CaseStatus
        location={
          Object {
            "pathname": "/case-status/",
            "state": Object {
              "details": Object {
                "action": "Register",
                "icon": "fa fa-check",
                "id": 12,
                "status": "Pending",
              },
            },
          }
        }
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "location": Object {
          "pathname": "/case-status/",
          "state": Object {
            "details": Object {
              "action": "Register",
              "icon": "fa fa-check",
              "id": 12,
              "status": "Pending",
            },
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CaseStatus
          location={
            Object {
              "pathname": "/case-status/",
              "state": Object {
                "details": Object {
                  "action": "Register",
                  "icon": "fa fa-check",
                  "id": 12,
                  "status": "Pending",
                },
              },
            }
          }
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "location": Object {
            "pathname": "/case-status/",
            "state": Object {
              "details": Object {
                "action": "Register",
                "icon": "fa fa-check",
                "id": 12,
                "status": "Pending",
              },
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
